lib/TPP/Dialect/Xsmm/XsmmUtils.cpp

```diff
@@ -148,6 +148,28 @@ FailureOr<BinaryInfo> getBinaryInfo(Value lhs, BinaryFlags lhsFlag, Value rhs,
   return binaryInfo;
 }
 
+FailureOr<int64_t> getLeadingDim(Type type, size_t pos) {
+  // Not shaped type, the leading dimension is the single scalar.
+  if (!isa<ShapedType>(type))
```

Comment:
This is unsafe. You check for `ShapedType` but cast to `MemrefType`.

A much simpler pattern is:
```c++
  auto memref = dyn_cast<MemRefType>(type);
  if (!memref)
    return 1;
  // Continue assuming it is a memref
```